### PR TITLE
Rename catboost

### LINKED ISSUE #737: Rename Catboost models
### 🚀 Feature Request

Currently we have some inconsistencies with names of the models. For example:
* `CatBoostModelMultiSegment`
* `SklearnMultiSegmentModel`
* `LinearMultiSegmentModel`
* `ElasticMultiSegmentModel`

It will be better to rename 
* `CatBoostModelMultiSegment` -> `CatBoostMultiSegmentModel`
* `CatBoostModelPerSegment` -> `CatBoostPerSegmentModel`

But this rename will break backward compatibility. To solve this we can rename the classes and just assign new classes to old names for compatibility.

### Motivation

Look above.

### Proposal

1. Rename `CatBoostModelMultiSegment` -> `CatBoostMultiSegmentModel`
2. Rename `CatBoostModelPerSegment` -> `CatBoostPerSegmentModel`
3. Check that we use new names in examples, readme, tests and documentation.
4. Assign new classes to old names and make sure that they can be imported.
5. Add deprecation warning that old names will be removed in version 2.0.

### Test cases

Make sure all tests passes and only new classes are used. 

You can check it by temporarily raising an error in old classes.

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [ ] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by julia-shenshina**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Closing issues
closes #737

### CODE REVIEW

File: etna/models/catboost.py [hunk 1]
```diff
@@ -113,7 +114,7 @@ class CatBoostModelPerSegment(PerSegmentModel):
     >>> future = ts.make_future(horizon)
     >>> model = CatBoostModelPerSegment()
```
Comment (Mr-Geekman) [inline etna/models/catboost.py:115]: Вот тут должна быть новая модель.

File: etna/models/catboost.py [hunk 2]
```diff
@@ -235,7 +236,7 @@ class CatBoostModelMultiSegment(MultiSegmentModel):
     >>> future = ts.make_future(horizon)
     >>> model = CatBoostModelMultiSegment()
```
Comment (Mr-Geekman) [inline etna/models/catboost.py:237]: Same about new name

File: etna/models/catboost.py (hunk 1 above)
Comment (julia-shenshina) [reply etna/models/catboost.py:115]: why? 

examples for class usage should be about the class they rely to, shouldn't they?
I added a warning that refers to a new class to docs and corresponding example to a new class 🤔

Comment (julia-shenshina) [reply etna/models/catboost.py:115]: we don't know if etna==2.0 comes soon 
so I would love to save full compatibility with old catboost models before it happens

Comment (Mr-Geekman) [reply etna/models/catboost.py:115]: It is a docstring example to a new class. Why should we use here and old one?
I think that example docstring for the old class should be in the old class.

Comment (julia-shenshina) [reply etna/models/catboost.py:115]: sorry, I was sure it was about old one
ok

File: etna/models/catboost.py (hunk 2 above)
Comment (julia-shenshina) [reply etna/models/catboost.py]: fixed

File: etna/models/catboost.py (hunk 1 above)
Comment (julia-shenshina) [reply etna/models/catboost.py]: fixed

**Review by martins0n — CHANGES_REQUESTED** [review]
You should add record to CHANGELOG.md

**Review by Mr-Geekman — APPROVED** [review]

**Review by martins0n — APPROVED** [review]

**Pull request merged**